compiler/rustc_mir_build/src/builder/expr/into.rs

```diff
@@ -244,6 +248,29 @@ impl<'a, 'tcx> Builder<'a, 'tcx> {
             ExprKind::LoopMatch { state, region_scope, ref arms, .. } => {
                 // FIXME add diagram
 
+                let dropless_arena = rustc_arena::DroplessArena::default();
+                let typeck_results = this.tcx.typeck(this.def_id);
+                let lint_level = this.tcx.local_def_id_to_hir_id(this.def_id);
```

Comment:
There is a commented out `lint_level` field in `LoopMatch`. That one should probably be uncommented and used here instead. The current code would use the lint level for the whole function. Alternatively adding a FIXME would be fine.